docs/concepts/workloads/vm.md

```diff
@@ -315,7 +449,74 @@ The above data does _not_ represent the _observed_ network configuration of the
+|------|-------------|
+| `status.storage.committed` | The total storage space committed to this `VirtualMachine`. |
+| `status.storage.uncommitted` | The total storage space potentially used by this `VirtualMachine` |
+| `status.storage.unshared` | The total storage space occupied by this VirtualMachine that is not shared with any other `VirtualMachine`. |
+
+For example, the following illustrates the storage usage for a `VirtualMachine`:
+
+```yaml
+status:
+  storage:
+    committed: 6Gi
+    uncommitted: 7Gi
+    unshared: 2Gi
+```
+
+Individual volume usage may be determined by inspecting [`status.volumes`](#volume-status).
+
+
+### Storage Quota
+
+`VirtualMachine` resources on a vSphere Supervisor must adhere to the namespace's storage quota:
+
+#### Storage Quota Reservation
+
+Deploying a `VirtualMachine` resource will fail an admission check unless the total capacity required by the disk(s) from the `VirtualMachineImage` or `ClusterVirtualMachineImage` is less than or equal to the amount of free space available in the namespace according to its storage quota. For example, consider the following:
+
+* A VM is deployed from an image that has a disk which is only `2Gi` in size, but has the capacity to grow to `100Gi`.
+* The storage quota for the namespace is `500Gi` and there is currently only `50Gi` available.
+
+The VM in the above scenario will fail the admission check. While the image's disk is only `2Gi` in size and there are `50Gi` of free space in the namespace, the disk's _total_ capacity is `100Gi`, which exceeds the quota.
+
+#### Storage Quota Usage
+
+Deployed `VirtualMachine` resources contribute to the total storage usage in a given namespace. The usage of a `VirtualMachine` is not calculated based on the total possible space the VM _can_consume, rather usage is the total number of unshared bytes on disk used by a `VirtualMachine` less the space used by any of the VM's [managed volumes](#volume-type). For example, consider the following:
+
+* A VM is using `2Gi` of unshared space.
+* The VM has three volumes, two of which are managed that, combined, consume `1Gi`.
+
+In the above scenario, the VM's total usage will be reported as `1Gi`.
+
+
+### Volumes
+
+A `VirtualMachine` resource's disks are referred to as _volumes_.
+
+#### Volume Type
+
+There are two types of volumes: _Managed_ and _Classic_:
+
+* **Managed** volumes refer to storage that is provided by [PersistentVolumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes).
+* **Classic** volumes refer to the disk(s) that came from the `VirtualMachineImage` or `ClusterVirtualMachineImage` from which the `VirtualMachine` was deployed.
+
+#### Adding Volumes
+
+##### Adding Classic Volumes
+
+There is no support for adding classic vSphere disks directly to `VirtualMachine` resources. Additional storage _must_ be provided by managed volumes, i.e. [PersistentVolumeClaims](https://kubernetes.io/docs/concepts/storage/persistent-volumes).
```

Comment:
Use https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims to link the PVCs?